plans/TOOLING.md

```diff
@@ -0,0 +1,241 @@
+# THIS IS A WORK IN PROGRESS
+
+This document describes the tentative roadmap for AsyncAPI tooling in the next ~6 months and the strategies we'll follow to create the tools.
+
+## Roadmap
+We want to start developing tools as we make progress with version 2.0.0. To minimize the impact of future specification changes in the tools (_just those for v2 while it's in development_), we'll perform [all the breaking changes of the spec](https://github.com/asyncapi/asyncapi/issues?utf8=%E2%9C%93&q=label%3A%22v2+breaking+change%22+) first, and then we'll start with tooling **development**. Notice I've used the word _development_ because I want to make it clear that we can start other types of work beforehand, like planning, architecting, documenting the approach, etc. Actually, this document is already a pre-development work.
+
+### Milestones
+
+#### :arrow_right: Create a Parser/Validator
+The first tool we would have to implement is the AsyncAPI parser/validator. The parser/validator is a core piece of all the other tooling we'll build in the future. Therefore it's a top priority for us.
+
+##### Requirements:
+1. It MUST support YAML and JSON as input formats.
+2. It MUST support the following schema formats: OpenAPI schemas, JSON Schema Draft 04-07, Protobuf, and Avro.
+3. It MUST support plugging in a new format parser.
+4. It MUST be implemented as a Go package and [exported as a C shared object](https://medium.com/learning-the-go-programming-language/calling-go-functions-from-other-languages-4c7d8bcc69bf).
```

Comment:
What is the skill level for Go within the contributor-base?

I like the idea of a core implementation with wrappers, and Go may be a good choice. There are JSON-Schema implementations (http://json-schema.org/implementations.html#validator-go) which may help us.

Has, however, Rustlang been considered? It can be made provably safe, doesn't have Go's package -dependency-on-github problems and generates significantly smaller code (Hello World is 2.0Mb in Go, 1.2Mb stripped, in Rust 251Kb/187Kb)?

To be fair, Rust may lack known JSON Schema validators.

Either language would be a learning opportunity (!) for me.

Reply:
Actually, there are several Rust crates which look interesting...

https://github.com/search?q=rust+json+schema
https://github.com/search?q=rust+avro
https://github.com/search?q=rust+protobuf

Reply:
I just started learning Golang a couple of days ago. I think the main reason for using Golang is the capability to generate C shared libraries: many languages have a way to call native C libraries.

Reply:
My experience with Go is short. I think Go is a good choice for performance reasons and it's also having an increasing adoption at companies, as well as a great community.

Not saying with this that Rust is not a great language. Probably is, I don't know 😄. And that's precisely my concern, that I haven't heard much of it yet.

Reply:
Thanks @fmvilas @rmelian @derberg one thing I wanted to mention in yesterday's call, but we ran out of time, is that I'm happy to Go with Go :smile: but I feel raising the issue of an alternative helped to clarify and explain and document the decision-making process behind the choice of backend language, and I think that's a good thing.

Reply:
haha! I hope we don't get rusty going with Go.

_(Someone stop this bad jokes hell, please)_